### PR TITLE
[MM-60066] Convert ./components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx from Class Component to Function Component

### PR DESCRIPTION
**Pull request by Camillarhi**
#### Summary

Convert class component to function component

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/27888
Jira https://mattermost.atlassian.net/browse/MM-60066
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx [hunk 1]
```diff
@@ -43,42 +43,25 @@ type Props = {
...
-            user,
-            channel,
-            isMuted,
-            actions: {
-                updateChannelNotifyProps,
-            },
-        } = this.props;
-
-        updateChannelNotifyProps(user.id, channel.id, {
+export default function MenuItemToggleMuteChannel({id, isMuted, channel, user, actions}: Props) {
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx:46]: Can we put each prop in a different line?
```suggestion
export default function MenuItemToggleMuteChannel({
    id,
    isMuted,
    channel,
    user,
    actions
}: Props) {
```

File: webapp/channels/src/components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx [hunk 2]
```diff
@@ -43,42 +43,25 @@ type Props = {
...
+    const handleClick = () => {
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx:47]: This is passed as a prop, so let's use a `useCallback`.

File: webapp/channels/src/components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx [hunk 3]
```diff
@@ -43,42 +43,25 @@ type Props = {
...
-        return (
-            <Menu.ItemAction
-                id={id}
-                onClick={this.handleClick}
-                text={text}
-            />
-        );
+    let text;
+    if (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL) {
+        text = isMuted ? localizeMessage('channel_header.unmuteConversation', 'Unmute Conversation') : localizeMessage('channel_header.muteConversation', 'Mute Conversation');
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx:55]: We are trying to get rid of `localizeMessage` so do you mind using intl here? (`useIntl` and `intl.formatMessage`)

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just some minor comments.

File: webapp/channels/src/components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx [hunk 4]
```diff
@@ -43,42 +43,36 @@ type Props = {
...
-            actions: {
-                updateChannelNotifyProps,
-            },
-        } = this.props;
+export default function MenuItemToggleMuteChannel({
+    id,
+    isMuted,
+    channel,
+    user,
+    actions}: Props) {
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx:51]: Small nit, just to be the same in all the codebase (yes, I should find a linting rule for this :sweat_smile: )
```suggestion
    actions,
}: Props) {
```

**Review by larkox — APPROVED** [review]
Great work! Just one minor thing, but I am approving it early.

File: webapp/channels/src/components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx [hunk 5]
```diff
@@ -43,42 +43,37 @@ type Props = {
...
-        }
-
-        return (
-            <Menu.ItemAction
-                id={id}
-                onClick={this.handleClick}
-                text={text}
-            />
-        );
+    }, [actions]);
```
Comment (Rajat-Dabade) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx:59]: Should `isMuted`, `user.id`, and `channel.id` be added as the dependency for `useCallback`? So to ensure the function is updated correctly when any of these values change.

**Review by Rajat-Dabade — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**